### PR TITLE
convert topology vm sidepanel details to use dynamic extensions

### PR DESCRIPTION
**Pull request by sahil143**
fixes: story: https://issues.redhat.com/browse/ODC-5846, https://issues.redhat.com/browse/ODC-6245. task: https://issues.redhat.com/browse/ODC-6110

This PR refactors the topology VM side panel to use dynamic extension and migrate the vm actions to use new action extension

image

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/actions/provider.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  } = vm;
+  const [vmi] = useK8sWatchResource<VMIKind>({
+    kind: kubevirtReferenceForModel(VirtualMachineInstanceModel),
+    name,
+    namespace,
+    isList: false,
+  });
+  const vmStatusBundle = useVMStatus(name, namespace);
+  const actions = React.useMemo(() => {
+    return vmStatusBundle
```
Comment (glekner) [inline frontend/packages/kubevirt-plugin/src/actions/provider.ts:32]: this whole section is not very readable. those spreads can cause performance issues, can you create a dictionary wrapped with memo if needed?
we also need to consider the new `printableStatus` that the VM yaml has.
you can check if the `printableStatus` is available like this.

https://github.com/openshift/console/pull/9761/files#diff-2edf5c3c0f283a0fe7f301625abdeb74011fe199291674e0119379b3100415eeR106

The available statuses as of now are
https://github.com/kubevirt/kubevirt/pull/5528/files#diff-5469726ef6ef072d9258db4593eb8811363099d9181a4e5dbd119af7b2b1414eR1137

**Review by glekner — COMMENTED** [review]
Amazing work @sahil143 !
left one comment on the provider

File: frontend/packages/kubevirt-plugin/src/actions/provider.ts (hunk 1 above)
Comment (sahil143) [reply frontend/packages/kubevirt-plugin/src/actions/provider.ts:32]: I don't understand how a dictionary would help here, action provider expects an array of actions.

Also, Do we have a hook that returns these statuses similar to `useVMStatus`, or can I get these from the status of VMKind resource when flag `FLAG_KUBEVIRT_HAS_PRINTABLESTATUS` is true?

**Comment by glekner** [comment]
the new status in the VM Table was in order to improve the performance of the table but in this context we don't have a table, so its ok to use the current status.

/lgtm

**Comment by sahil143** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Comment by rohitkrai03** [comment]
/label qe-approved
Verified on cluster created using cluster bot. Everything works as expected.

**Pull request merged**